src/main/java/tech/gruppone/stalker/server/model/ConnectedOrganization.java

```diff
@@ -0,0 +1,42 @@
+package tech.gruppone.stalker.server.model;
+
+import org.springframework.data.annotation.Id;
+
+import lombok.NonNull;
+
+// Model created to reflect exactly keys of the json which api receives by app and web app
+public class ConnectedOrganization {
```

Comment:
usare lombok `@Value`

questo credo lo chiamerei OrganizationData (se è quello che ho capito che sia). e in questo caso non serve la proprietà Id perché l'unico contesto in cui dovremmo usarlo direttamente è la creazione di una nuova org, che quindi è un INSERT dove l'id viene generato automaticamente

Reply:
Ok per la ridenominzazione.
Però questa classe è stata creata per l'endpoint /user/{id}/organizations/connections, che fa:
_Get the list of organizations that a user has connected to._

In realtà basterebbe esplicitare solo l'id nel json che andiamo a creare, ma in questa classe ho anche preferito inserire il nome e la descrizione associata a quell'organizationID.

Reply:
> Però questa classe è stata creata per l'endpoint /user/{id}/organizations/connections, che fa:
> _Get the list of organizations that a user has connected to._

ma se ritorna la lista perché questo contiene una sola istanza? se serve a quello mi aspetterei una classe con una singola proprietà:

`List<Organization> organizations`

ma dato che il response di quella richiesta dovrebbe contenere solo gli id delle organizzazioni a cui l'utente è collegato, la proprietà dovrebbe essere:

`List<Long> connectedOrganizations`

> In realtà basterebbe esplicitare solo l'id nel json che andiamo a creare, ma in questa classe ho anche preferito inserire il nome e la descrizione associata a quell'organizationID (perchè un id solo soletto così significa poco)

significa esattamente quello che abbiamo deciso, e cioè che quell'endpoint ritorna un oggetto fatto in quel modo.

Comunque fare una classe apposta nel modello mi sembra ridondante. dato che questo tipo di risposta viene data SOLO da questo endpoint, farei o un interfaccia + classe anonima o una static inner class. qua chiederei che ne pensa a @RiccardoAgatea . Altra soluzione potrebbe essere mettere un subpackage model.responses e mettere la classe (e le altre che verranno) la dentro

Reply:
Comunque posso sempre modificarlo eh